src/EditorFeatures/Core.Wpf/InlineRename/CommandHandlers/RenameCommandHandler_RenameHandler.cs

```diff
@@ -42,19 +43,15 @@ public CommandState GetCommandState(RenameCommandArgs args, Func<CommandState> n
                 return nextHandler();
             }
 
-            return CommandState.Available;
+            return VisualStudio.Commanding.CommandState.CommandIsAvailable;
         }
 
-        public void ExecuteCommand(RenameCommandArgs args, Action nextHandler)
+        public void ExecuteCommand(RenameCommandArgs args, Action nextHandler, CommandExecutionContext context)
         {
-            _waitIndicator.Wait(
-                title: EditorFeaturesResources.Rename,
-                message: EditorFeaturesResources.Finding_token_to_rename,
-                allowCancel: true,
-                action: waitContext =>
+            using (context.WaitContext.AddScope(allowCancellation: true, EditorFeaturesResources.Finding_token_to_rename))
             {
-                ExecuteRenameWorker(args, waitContext.CancellationToken);
-            });
+                ExecuteRenameWorker(args, context.WaitContext.CancellationToken);
```

Comment:
i'm tring to understand the relation between allowCancellation and .CancellationToken.  Is it a stack?  What happens if some other code sets allowCancellation to false?  Will that affect our cancelation token?

Reply:
I renamed it to UserCancellation token, does it look more readable now?
`ExecuteRenameWorker(args, context.WaitContext.UserCancellationToken);`